backend/src/test/java/com/benine/backend/http/CameraStreamHandlerTest.java

```diff
@@ -0,0 +1,112 @@
+package com.benine.backend.http;
+
+import com.benine.backend.camera.ipcameracontrol.IPCamera;
+import com.benine.backend.video.MJPEGStreamReader;
+import com.benine.backend.video.Stream;
+import com.benine.backend.video.StreamNotAvailableException;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import javax.servlet.ServletException;
+
+import static org.mockito.Mockito.*;
+
+/**
+ * Created on 4-5-16.
+ */
+public class CameraStreamHandlerTest extends CameraRequestHandlerTest {
+
+  IPCamera cam = mock(IPCamera.class);
+
+  Stream stream;
+  MJPEGStreamReader streamReader;
+  Thread streamReaderThread;
+
+  @Override
+  public CameraRequestHandler supplyHandler() {
+    return new CameraStreamHandler();
+  }
+
+  @Before
+  public void initialize() throws IOException {
+    super.initialize();
+    when(cameracontroller.getCameraById(42)).thenReturn(cam);
+
+    stream = mock(Stream.class);
+    when(stream.getInputStream()).thenReturn(new BufferedInputStream(new FileInputStream("resources" + File.separator + "test" + File.separator + "testmjpeg.mjpg")));
+
+    streamReader = new MJPEGStreamReader(stream);
+    //when(streamReader.getBoundary()).thenReturn("--testBoundary");
+
+    doThrow(new IOException()).when(httpresponseMock).getOutputStream();
+    try {
+      when(streamController.getStreamReader(42)).thenReturn(streamReader);
+      streamReaderThread = new Thread(streamReader);
+      streamReaderThread.start();
+    } catch (StreamNotAvailableException e) {
+      e.printStackTrace();
+    }
+  }
+
```

Comment:
No this is correct, since a thread is needed in most of the tests this is a separate method creating the thread.